src/command_modules/azure-cli-policyinsights/azure/cli/command_modules/policyinsights/_params.py

```diff
@@ -0,0 +1,153 @@
+            help='Resource group name.')
+        c.argument(
+            'resource_id',
+            options_list=('--resource-id', '-r'),
+            help='Resource id.')
+        c.argument(
+            'policy_set_definition_name',
+            options_list=('--policy-set-definition-name', '-ps'),
+            completer=get_policy_set_completion_list,
+            help='Policy set definition name.')
+        c.argument(
+            'policy_definition_name',
+            options_list=('--policy-definition-name', '-pd'),
+            completer=get_policy_completion_list,
+            help='Policy definition name.')
+        c.argument(
+            'policy_assignment_name',
+            options_list=('--policy-assignment-name', '-pa'),
+            completer=get_policy_assignment_completion_list,
+            help='Policy assignment name.')
+        c.argument(
+            'from_value',
+            options_list=('--from'),
+            help='Timestamp specifying the start time of the interval to query.')
+        c.argument(
+            'to_value',
+            options_list=('--to'),
+            help='Timestamp specifying the end time of the interval to query.')
+        c.argument(
+            'top_value',
+            options_list=('--top'),
+            type=int,
+            help='Maximum number of records to return.')
+        c.argument(
+            'order_by_clause',
+            options_list=('--order-by'),
+            help='Ordering expression using OData notation.')
+        c.argument(
+            'select_clause',
+            options_list=('--select'),
+            help='Select expression using OData notation.')
+        c.argument(
+            'filter_clause',
+            options_list=('--filter'),
+            help='Filter expression using OData notation.')
+        c.argument(
+            'apply_clause',
+            options_list=('--apply'),
+            help='Apply expression for aggregations using OData notation.')
+
+    with self.argument_context('policy state') as c:
+        c.argument(
+            'all_results',
+            options_list=('--all'),
+            arg_type=get_three_state_flag(),
+            help='Within the specified time interval, get all policy states instead of the latest only.')
+        c.argument(
+            'management_group_name',
+            options_list=('--management-group-name', '-m'),
+            help='Management group name.')
+        c.argument(
+            'subscription_id',
+            options_list=('--subscription-id', '-s'),
+            help='Subscription id.')
+        c.argument(
+            'resource_group_name',
+            options_list=('--resource-group-name', '-g'),
+            arg_type=resource_group_name_type,
+            completer=get_resource_group_completion_list,
+            help='Resource group name.')
+        c.argument(
+            'resource_id',
+            options_list=('--resource-id', '-r'),
```

Comment:
This makes the CLI inconsistent with the PS API, and ambiguous in itself. Our RP is an extension RP, and can be called with any resource as a scope. In the swagger API we specifically refrained from making parts of ARM resource ID separate parameters because that would make it impossible to support multiple nested level resources. API board review agreed on that. If we change this now in the CLI contract (so that we can support resource names), we will need to add multiple levels of namespace and resource type arguments, which makes the API unusable (and inconsistent with the PS API. Feel free to reach out to me to go over the swagger specs, PS API and any other details.

Reply:
Consistency with Powershell is not important in the CLI, but consistency with other modules in the CLI is. This requires no change to swagger or the API.

Reply:
As discussed offline, I'll implement additional parameters to make resourceid a resource that can take either an ID or a name.